### DISCUSSION TITLE
Use justfile instead of precommit

### DISCUSSION THREAD

**Discussion by lesnik512** [community-of-python/microbootstrap#49] [Ideas]
Why do we need precommit here?
I see big disadvantages of it:
- versions are not consistent with pyproject
- separate virtual environments -> slower work, not obvious behaviour

What do you think about switching to justfile commands? As in https://github.com/modern-python/that-depends/blob/main/Justfile

@vrslev @insani7y

**Comment by lesnik512** [comment]
And here some argument against precommit also https://github.com/community-of-python/microbootstrap/issues/39

**Comment by lesnik512** [comment]
And pre-commit still can be used to run justfile command
https://github.com/modern-python/that-depends/blob/main/.pre-commit-config.yaml

Here I applied it to check https://github.com/community-of-python/microbootstrap/pull/48

**Comment by vrslev** [comment]
I'm "pro" justfile :)

**Comment by lesnik512** [comment]
I've created separated workflow with `just` for now as a proof of concept and applied it here.

**Reply by vrslev** [reply]
Nice!
